src/main/java/heronarts/lx/buffer/FloatTensorAddBlend.java

```diff
@@ -0,0 +1,161 @@
+package heronarts.lx.buffer;
+
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.indexing.NDArrayIndex;
+import org.nd4j.linalg.ops.transforms.Transforms;
+
+public class FloatTensorAddBlend extends BaseTensorBlend {
+
+  /**
+   * Blends the src buffer onto the destination buffer at the specified alpha amount.
+   * All tensors must be 2D with shape [N, 4] and FLOAT data type with values in range [0, 1].
+   *
+   * @param dst Destination buffer (lower layer) - FLOAT tensor [N, 4] with values [0, 1]
+   * @param src Source buffer (top layer) - FLOAT tensor [N, 4] with values [0, 1]
+   * @param alpha Alpha blend, from 0-1
+   * @param output Output buffer - FLOAT tensor [N, 4], may be the same as src or dst
+   * @param start Starting index to blend
+   * @param num Number of pixels to blend
+   */
+  @Override
+  public void blend(INDArray dst, INDArray src, double alpha, INDArray output, int start, int num) {
+    // Validate inputs
+    validateInputs(DataType.FLOAT, dst, src, output);
+
+    // Work on slices
+    INDArray dstSlice = dst.get(NDArrayIndex.interval(start, start + num), NDArrayIndex.all());
+    INDArray srcSlice = src.get(NDArrayIndex.interval(start, start + num), NDArrayIndex.all());
+    INDArray outSlice = output.get(NDArrayIndex.interval(start, start + num), NDArrayIndex.all());
+
+    // Copy dst to output if they're different objects
+    if (output != dst) {
+      outSlice.assign(dstSlice);
+    }
+
+    // Extract channels
+    INDArray outAlpha = outSlice.getColumn(0);
+    INDArray outR = outSlice.getColumn(1);
+    INDArray outG = outSlice.getColumn(2);
+    INDArray outB = outSlice.getColumn(3);
+
+    INDArray srcAlpha = srcSlice.getColumn(0);
+    INDArray srcR = srcSlice.getColumn(1);
+    INDArray srcG = srcSlice.getColumn(2);
+    INDArray srcB = srcSlice.getColumn(3);
+
+    // Convert to integer space for exact calculation, then back to float
+    // This ensures we match the UINT8 implementation exactly
+
+    // Scale [0,1] to [0,255] for integer math
+    INDArray srcAlpha255 = srcAlpha.mul(255.0);
+    INDArray srcR255 = srcR.mul(255.0);
+    INDArray srcG255 = srcG.mul(255.0);
+    INDArray srcB255 = srcB.mul(255.0);
+
+    // Apply the exact integer formula
+    double alphaScale = alpha * 256.0;
+    INDArray effectiveAlpha = srcAlpha255.mul(alphaScale).div(256.0);
+
+    // Add rounding: effectiveAlpha += (effectiveAlpha >= 127.5 ? 1 : 0)
+    INDArray roundingMask = effectiveAlpha.gte(127.5);
+    effectiveAlpha.addi(roundingMask.castTo(DataType.FLOAT));
+
+    // Additive blend in 255 space: out = dst + (src * effectiveAlpha / 256)
```

Comment:
For this float implementation I would strip out all this 255/256 noise and do the guts of the compute in 0-1 space.